src/module/skb_tracking/event.rs

```diff
@@ -29,13 +32,9 @@ pub(crate) struct SkbTrackingEvent {
 }
 
 impl RawEventSectionFactory for SkbTrackingEvent {
-    fn from_raw(&mut self, mut raw_sections: Vec<BpfRawSection>) -> Result<Box<dyn EventSection>> {
-        if raw_sections.len() != 1 {
-            bail!("Skb tracking event from BPF must be a single section");
-        }
-
-        // Unwrap as we just checked the vector contains 1 element.
-        let raw = parse_raw_section::<BpfTrackingEvent>(&raw_sections.pop().unwrap())?;
+    fn from_raw(&mut self, raw_sections: Vec<BpfRawSection>) -> Result<Box<dyn EventSection>> {
```

Comment:
OK, not for `KernelEvent` and `UserspaceEvent`, although I'd argue that we should define a struct and use Plain even if it's a temporary one and we then need to do some transformations instead of accessing the fields by offset which I consider really ugly and hard to maintain. But CommonEvent can benefit from this directly

Reply:
100% we should not do that in this PR :-D